src/traits.rs

```diff
@@ -55,6 +55,9 @@ pub trait Object<'data, 'file> {
     /// Get an iterator over the sections in the file.
     fn sections(&'file self) -> Self::SectionIterator;
 
+    /// Get the debugging symbol at the given index.
+    fn symbol_by_index(&self, index: u64) -> Option<Symbol<'data>>;
```

Comment:
I think we should clarify in the doc comment that this index is the n<sup>th</sup> symbol defined in the file, *not* an offset into the file's bytes. Easy for consumers to make this mistake.